api/views/workflow.py

```diff
@@ -48,14 +49,13 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     ordering_fields = ('long_title', )
     serializer_class = SourceSerializer
     permission_classes = [DjangoModelPermissionsOrAnonReadOnly, ]
+    filter_backends = (DjangoFilterBackend,)
+    filter_fields = ('long_title',)
 
-    queryset = Source.objects.none()  # Required for DjangoModelPermissions
+    queryset = Source.objects.exclude(icon='').exclude(is_deleted=True)  # Required for DjangoModelPermissions
```

Comment:
Would you mind adding a comment (or description?) on the `icon` field definition that explains how sources without icons aren't exposed in the api/front end? It's unintuitive, easy to forget, and unwritten besides this exclude.